clientconn.go

```diff
@@ -119,6 +132,7 @@ type ClientConn struct {
 
 func (cc *ClientConn) resetTransport(closeTransport bool) error {
 	var retries int
+	totime := time.Now().Add(cc.dopts.timeout)
```

Comment:
It should come out to time.Now(), given the zero value for timeout. However, we check down below that timeout > 0 before running through the timeout check.

Perhaps it would be better written as
var totime time.Time
if cc.dopts.timeout > 0 {
  totime := time.Now().Add(cc.dopts.timeout)
}
...
  if !totime.IsZero() && otherTimeoutCheck {
      // etc
  }

Would you prefer that approach?